backend/src/agents/tools/auth_tools.py

```
@@ -0,0 +1,537 @@
+        # res will be a Python object deserialized from JSON
+        if isinstance(res, dict):
+            if res.get("__error") is not None:
+                return False
+            if res.get("__status") == 401:
+                return False
+            # likely a user object if not status/error
+            if res.get("user") or res.get("email") or res.get("username") or res.get("bio"):
+                return True
+            # some APIs return the object itself
+            # if it contains at least one key other than our markers, consider authenticated
+            if any(k for k in res.keys() if not k.startswith("__")):
```

Ruff (0.12.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)